### PR TITLE
feat: implement grpc-observability plugin

### PR DESCRIPTION
**Pull request by lidizheng**
This PR is the draft of what grpc-observability plugin would look like. At this stage, we don't have a concrete spec for logging, nor configurations. This PR focuses on the workflow of: opt-in, parse config, log basic start/end events, export to CloudLogging.

Example usage:

```go
import "google.golang.org/grpc/observability"

func main() {
    observability.Start()
    defer observability.End()
    // Use gRPC as normal
}
```

This PR adds 4 new APIs to public surface under `google.golang.org/grpc/stats/defaults`:

```go
func SetDefaultClientStatsHandler(handler stats.Handler)
func GetDefaultClientStatsHandler() stats.Handler
func SetDefaultServerStatsHandler(handler stats.Handler)
func GetDefaultServerStatsHandler() stats.Handler
```

The above APIs can be used by any 3rd-party library, if they want to do something advanced about stats. I will write up an gRFC if Go team is okay with the direction. If not, we can discuss how to improve it.

PS. I just heard today that there is a similar effort named DialOption interceptors. But what about server-side?

---

This PR also adds a basic test case for unary call. The correctness of the telemetry data will be validated via unit tests. The interoperability between gRPC app and backends will be tested via a cross-language end2end test.

I manually tested that the CloudLogging exporter works.

---

As mentioned... the spec is not settled yet, so I included the entire proto file. What's the best practice for proto generated code for grpc-go?

### CODE REVIEW

**Comment by lidizheng** [comment]
cc @dfawley

**Review by dfawley — COMMENTED** [review]
I have some comments that I forgot to send on this.

**Comment by dfawley** [comment]
Generally, I'd prefer if you don't open PRs for discussions/previews/drafts.  For these things, you can create a PR or point to a commit in your fork instead.  Thanks!

**Comment by lidizheng** [comment]
My bad. I will point to a commit in future.

File: observability/go.mod [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module google.golang.org/grpc/observability
```
[the remark this answers is not in the document]
Comment (lidizheng) [reply observability/go.mod:1]: Alternatively, can we depend on cloud logging in the main go module?

File: observability/config/observability_config.proto [hunk 2]
```diff
@@ -0,0 +1,101 @@
+syntax = "proto3";
```
[the remark this answers is not in the document]
Comment (lidizheng) [reply observability/config/observability_config.proto:1]: I will move it, once we settled the logging config.

File: observability/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (dfawley) [reply observability/go.mod:1]: Hmm, good point, we probably don't want to add `cloud.google.com/go/logging` to our main module.

**Pull request closed without merging**